pkg/storage/odf.go

```diff
@@ -0,0 +1,279 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	odfoperatorv1alpha1 "github.com/red-hat-storage/odf-operator/api/v1alpha1"
+
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StorageSystemBuilder provides struct for storageSystem object containing connection
+// to the cluster and the storageSystem definitions.
+type StorageSystemBuilder struct {
+	// StorageSystem definition. Used to create a storageSystem object
+	Definition *odfoperatorv1alpha1.StorageSystem
+	// Created storageSystem object
+	Object *odfoperatorv1alpha1.StorageSystem
+	// api client to interact with the cluster.
+	apiClient goclient.Client
+	// Used in functions that define or mutate storageSystem definition. errorMsg is processed before the
+	// storageSystem object is created.
+	errorMsg string
+}
+
+// NewStorageSystemBuilder creates a new instance of Builder.
+func NewStorageSystemBuilder(apiClient *clients.Settings, name, nsname string) *StorageSystemBuilder {
```

Comment:
Same here. Please re-name to NewOdfSystemBuilder

Reply:
same here